pkg/ibmvpc/vpc.go

```diff
@@ -417,65 +417,65 @@ func (n *NACL) GeneralConnectivityPerSubnet(subnet *Subnet) string {
 	return res
 }
 
-func getNodeCidrs(n vpcmodel.Node) (subnetCidr, nodeCidr string, subnet *Subnet, err error) {
-	switch t := n.(type) {
+func subnetFromNode(node vpcmodel.Node) (subnet *Subnet, err error) {
+	switch concreteNode := node.(type) {
 	case *NetworkInterface:
-		return t.subnet.cidr, t.CidrOrAddress(), t.subnet, nil
+		return concreteNode.subnet, nil
 	case *IKSNode:
-		return t.subnet.cidr, t.CidrOrAddress(), t.subnet, nil
+		return concreteNode.subnet, nil
 	case *ReservedIP:
-		return t.subnet.cidr, t.CidrOrAddress(), t.subnet, nil
+		return concreteNode.subnet, nil
 	default:
-		return "", "", nil, fmt.Errorf("cannot get cidr for node: %+v", n)
+		return nil, fmt.Errorf("cannot get cidr for node: %+v", node)
 	}
 }
 
 func (n *NACL) initConnectivityComputation(src, dst vpcmodel.Node,
-	isIngress bool) (targetNode vpcmodel.Node, subnetCidr, inSubnetCidr string, subnet *Subnet, err error) {
+	isIngress bool) (targetNode, nodeInSubnet vpcmodel.Node, subnet *Subnet, err error) {
 	if isIngress {
-		targetNode = src
-		subnetCidr, inSubnetCidr, subnet, err = getNodeCidrs(dst)
+		targetNode, nodeInSubnet = src, dst
 	} else {
-		targetNode = dst
-		subnetCidr, inSubnetCidr, subnet, err = getNodeCidrs(src)
+		targetNode, nodeInSubnet = dst, src
 	}
-	return targetNode, subnetCidr, inSubnetCidr, subnet, err
+	subnet, err = subnetFromNode(nodeInSubnet)
+	return targetNode, nodeInSubnet, subnet, err
 }
 
 func (n *NACL) AllowedConnectivity(src, dst vpcmodel.Node, isIngress bool) (*common.ConnectionSet, error) {
-	targetNode, subnetCidr, inSubnetCidr, subnet, err := n.initConnectivityComputation(src, dst, isIngress)
+	targetNode, nodeInSubnet, subnet, err := n.initConnectivityComputation(src, dst, isIngress)
 	if err != nil {
 		return nil, err
 	}
 	// check if the subnet of the given node is affected by this nacl
-	if _, ok := n.subnets[subnetCidr]; !ok {
+	if _, ok := n.subnets[subnet.cidr]; !ok {
 		return vpcmodel.NoConns(), nil // not affected by current nacl
 	}
 	// TODO: differentiate between "has no effect" vs "affects with allow-all / allow-none "
-	if targetNode.IPBlock().ContainedIn(subnet.ipblock) {
+	if targetNode.IsInternal() && targetNode.IPBlock().ContainedIn(subnet.ipblock) {
```

Comment:
This is the place of the #344 bug - but after the changes, why is the check `targetNode.IsInternal()` needed?
`IPBlock` is well defined for external addresses

Reply:
Just to save a call to `ContainedIn` when not necessary

Reply:
perhaps add a comment, to avoid confusing

Reply:
added comment